From 43e9825e53dcee61522e805924c197131829ffd8 Mon Sep 17 00:00:00 2001
Message-Id: <43e9825e53dcee61522e805924c197131829ffd8.1424979244.git.feitong.yi@intel.com>
In-Reply-To: <d8466aca83b11ab8b0870e775536306782d73e9f.1424979244.git.feitong.yi@intel.com>
References: <d8466aca83b11ab8b0870e775536306782d73e9f.1424979244.git.feitong.yi@intel.com>
From: Arun Siluvery <arun.siluvery@linux.intel.com>
Date: Thu, 19 Feb 2015 11:36:01 +0000
Subject: [PATCH 26/28] SQUASHME! [VPG]: drm/i915: Make ring buffer mode WA
 init functions static
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

When a patch which initializes WA for CHV in ring buffer mode is pulled in
from upstream these functions were exported but they were static in
upstream patch. This patch fixes this difference.

This patch should be squashed into the following patch,

     Author: Ville Syrjälä <ville.syrjala@linux.intel.com>
     Date:   Wed Aug 27 17:33:12 2014 +0300
     Upstream-repo: git://anongit.freedesktop.org/drm-intel
     Upstream-commit: 00e1e623e62cd8452e28633182b91ddcbb70cc7c
     Change-Id: Ia7269ec6c7390b19484bec5c21386e4ca63270d1
     FROM_UPSTREAM [VPG]: drm/i915: Init some CHV workarounds via LRIs in ring->init_context()

Change-Id: Ia14e04f1206dd92df7e244bb7283248d0c6deedc
Signed-off-by: Arun Siluvery <arun.siluvery@linux.intel.com>
---
 drivers/gpu/drm/i915/intel_ringbuffer.c |    4 ++--
 drivers/gpu/drm/i915/intel_ringbuffer.h |    2 --
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_ringbuffer.c b/drivers/gpu/drm/i915/intel_ringbuffer.c
index 1d9af73..9a98990 100644
--- a/drivers/gpu/drm/i915/intel_ringbuffer.c
+++ b/drivers/gpu/drm/i915/intel_ringbuffer.c
@@ -1228,7 +1228,7 @@ static int wa_add(struct drm_i915_private *dev_priv,
 
 #define WA_WRITE(addr, val) WA_REG(addr, val, 0xffffffff)
 
-int bdw_init_workarounds(struct intel_engine_cs *ring)
+static int bdw_init_workarounds(struct intel_engine_cs *ring)
 {
 	struct drm_device *dev = ring->dev;
 	struct drm_i915_private *dev_priv = dev->dev_private;
@@ -1273,7 +1273,7 @@ int bdw_init_workarounds(struct intel_engine_cs *ring)
 	return 0;
 }
 
-int chv_init_workarounds(struct intel_engine_cs *ring)
+static int chv_init_workarounds(struct intel_engine_cs *ring)
 {
 	struct drm_device *dev = ring->dev;
 	struct drm_i915_private *dev_priv = dev->dev_private;
diff --git a/drivers/gpu/drm/i915/intel_ringbuffer.h b/drivers/gpu/drm/i915/intel_ringbuffer.h
index 3ff4450..368bdea 100644
--- a/drivers/gpu/drm/i915/intel_ringbuffer.h
+++ b/drivers/gpu/drm/i915/intel_ringbuffer.h
@@ -647,8 +647,6 @@ int intel_init_vebox_ring_buffer(struct drm_device *dev);
 u64 intel_ring_get_active_head(struct intel_engine_cs *ring);
 void intel_ring_setup_status_page(struct intel_engine_cs *ring);
 u32 get_pipe_control_scratch_addr(struct intel_engine_cs *ring);
-int bdw_init_workarounds(struct intel_engine_cs *ring);
-int chv_init_workarounds(struct intel_engine_cs *ring);
 int init_workarounds_ring(struct intel_engine_cs *ring);
 
 
-- 
1.7.9.5

